magpie/config.py

```diff
@@ -0,0 +1,96 @@
+import logging
+from typing import TYPE_CHECKING
+
+from jsonschema import validators
+
+from magpie.utils import get_logger, print_log
+from magpie.services import SERVICE_TYPE_DICT
+from magpie.cli.sync_services import SYNC_SERVICES_TYPES
+
+if TYPE_CHECKING:
+    from magpie.typedefs import JSON, ServicesConfig
+
+LOGGER = get_logger(__name__)
+
+
+SERVICE_HOOK_ITEM_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "type": {"type": "string", "enum": ["request", "response"]},
+        "path": {"type": "string"},
+        "query": {"type": "string", "default": r".*"},
+        "method": {"type": "string", "default": "*", "enum": [
+            "HEAD", "GET", "POST", "PUT", "PATCH", "DELETE",
+            "head", "get", "post", "put", "patch", "delete",
+            "*"
```

Comment:
Sadly we cannot since `AnyRequestMethod` exists only during `TYPE_CHECKING = 1` context when run by IDE/linters. 
We cannot do the other way around either, because `Literal` must be a list of literal strings (invalid to do something like `Literal[*list_values]`).